### PR TITLE
api plugin documentation

### PR DESCRIPTION
**Pull request by Caesarsage**
created a new docs.html file that has all zc_chess plugin backend documentations accessed from the endpoint  - /api/docs/v1

This will constantly be updated till all endpoint is complete , Please merge and do not close if possible

Issue #487

### CODE REVIEW

**Comment by NdubuisiJr** [comment]
No… we can’t use this..

**Comment by Caesarsage** [comment]
> No… we can’t use this..

state the reasons why please ?

**Comment by NdubuisiJr** [comment]
Documentation should be generated automatically based on available routes and code documentation…

Something like OpenApi spec or swagger

**Comment by Caesarsage** [comment]
> Documentation should be generated automatically based on available routes and code documentation…
> 
> Something like OpenApi spec or swagger

oh okay, True. I will use swagger UI then

**Comment by Caesarsage** [comment]
> > Documentation should be generated automatically based on available routes and code documentation…
> > Something like OpenApi spec or swagger
> 
> oh okay, True. I will use swagger UI then

check this out @NdubuisiJr for a start

**Comment by NdubuisiJr** [comment]
Okay
I'll look at it now

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]
Please answer these questions so we can continue?

File: .idea/inspectionProfiles/profiles_settings.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
```
[the remark this answers is not in the document]
Comment (Caesarsage) [reply .idea/inspectionProfiles/profiles_settings.xml:1]: these files nor folders are not even in my local system

**Pull request merged**